File: articles/calculus-for-systems-modeling/articles/climate-feedback-models/cpp/climate_feedback_models.cpp
#include <cmath>
#include <iostream>

double one_box(double forcing, double feedback, double heat_capacity, double time){
  double equilibrium = forcing / feedback;
  return equilibrium * (1.0 - std::exp(-(feedback / heat_capacity) * time));
}

int main(){
  double forcing=3.7, c=8.0;
  std::cout << "time,weak_feedback,baseline_feedback,strong_feedback\n";
  for(int t=0;t<=100;t+=10){
    std::cout << t << "," << one_box(forcing,0.9,c,t) << "," << one_box(forcing,1.2,c,t) << "," << one_box(forcing,1.6,c,t) << "\n";
  }
}

File: articles/calculus-for-systems-modeling/articles/climate-feedback-models/fortran/climate_feedback_models.f90
program climate_feedback_models
  implicit none
  integer :: t
  real(8) :: forcing, c, weak, baseline, strong
  forcing = 3.7d0
  c = 8.0d0
  print '(A)', 'time weak_feedback baseline_feedback strong_feedback'
  do t = 0, 100, 10
    weak = (forcing/0.9d0) * (1.0d0 - exp(-(0.9d0/c)*dble(t)))
    baseline = (forcing/1.2d0) * (1.0d0 - exp(-(1.2d0/c)*dble(t)))
    strong = (forcing/1.6d0) * (1.0d0 - exp(-(1.6d0/c)*dble(t)))
    print '(I0,1X,F12.6,1X,F12.6,1X,F12.6)', t, weak, baseline, strong
  end do
end program
